idl/sprot.idol

```diff
@@ -0,0 +1,72 @@
+// SP to RoT communications over SPI
+
+Interface(
+    name: "SpRot",
+    ops: {
+        "send_recv": (
+            doc: "Send then receive a message from the RoT",
+            args: {
+                "msgtype": (
+                    type: "MsgType",
+                    recv: FromPrimitive("u8"),
+                ),
+            },
+            leases: {
+                "message": (type: "[u8]", read: true),
```

Comment:
Do we want `max_len` caps on all the leases in this interface?

Reply:
The source of truth for the limit is in drv/sprot-api/src/lib.rs: PAYLOAD_SIZE_MAX which, for SP and RoT compiled in the same release, is the same value.
We could change this between releases and during RoT update, the SP and RoT may have a value for PAYLOAD_SIZE_MAX that is different from each other.
Normally, one would use the constants generated by idolatry to size the buffers in the implementation.
The RoT doesn't offer the SpRot idol client interface and doesn't have those idol constants handy.
I don't want to have two sources of truth to be maintained.
Should I still put limits here?

Reply:
This is an open question. I'll think about this for a future PR.